Makefile

```diff
@@ -1605,42 +1605,41 @@ llama-q8dot: pocs/vdot/q8dot.cpp ggml/src/ggml.o \
 # Mark legacy binary targets as .PHONY so that they are always checked.
 .PHONY: main quantize perplexity embedding server
 
+# Define the object file target
+examples/deprecation-warning/deprecation-warning.o: examples/deprecation-warning/deprecation-warning.cpp
+	$(CXX) $(CXXFLAGS) -c $< -o $@
+
 # NOTE: We currently will always build the deprecation-warning `main` and `server` binaries to help users migrate.
 #  Eventually we will want to remove these target from building all the time.
-main: examples/deprecation-warning/deprecation-warning.cpp
-	$(CXX) $(CXXFLAGS) -c $< -o $(call GET_OBJ_FILE, $<)
-	$(CXX) $(CXXFLAGS) $(filter-out $<,$^) $(call GET_OBJ_FILE, $<) -o $@ $(LDFLAGS)
+main: examples/deprecation-warning/deprecation-warning.o
+	$(CXX) $(LDFLAGS) $< -o $@
```

Comment:
This is probably just being cautious, but I'd keep CXXFLAGS here at the front and LDFLAGS at the end of the command like it was before. It may matter when cross-compiling for some weird platform.

Reply:
Thank you for the help!

If you can't tell, I'm not the strongest with makefiles -- if you feel like you'd rather open the PR, I would not be offended!

Reply:
Naah, if you fix this one thing I think it will be good to go.